tests/integration/fixtures/processing.py

```diff
@@ -167,6 +167,31 @@ def poll(self, timeout=None):
             timeout = self.timeout
         return self.consumer.poll(timeout=timeout)
 
+    def poll_many(self, timeout=None, n=None):
+        if timeout is None:
+            timeout = self.timeout
+
+        if n == 0:
+            self.assert_empty()
+            return
+
+        messages = 0
+
+        # Wait for the first outcome to show up for the full timeout duration
+        message = self.poll(timeout)
+        while message is not None:
+            yield message
+            messages += 1
+
+            if messages == n:
+                self.assert_empty()
+                break
+
+            message = self.poll(min(2, timeout) if n is None else timeout)
```

Comment:
Should the 2 be configurable? E.g. `def poll_many(self, initial_timeout=None, subsequent_timeout=2, n=None)`?

Reply:
Yes, I haven't done that yet because I couldn't decide on good names.